yang/frr-interface.yang

```diff
@@ -24,6 +36,230 @@ module frr-interface {
       "Initial revision.";
   }
 
+  identity other {
+    base ietf-if:interface-type;
+    description
+      "Other type";
+  }
+
+  identity unknown {
+    base ietf-if:interface-type;
+    description
+      "Unknown type";
+  }
+
+  identity ethernet {
+    base ietf-if:interface-type;
+    description
+      "Ethernet type";
+  }
+
+  identity exper-ethernet {
+    base ietf-if:interface-type;
+    description
+      "Experimental Ethernet type";
+  }
+
+  identity loopback {
+    base ietf-if:interface-type;
+    description
+      "Loopback type";
+  }
+
+  identity pimreg {
+    base ietf-if:interface-type;
+    description
+      "PIMSM Registration.";
+  }
+
+  identity ipip {
+    base ietf-if:interface-type;
+    description
+      "IPIP Tunnel.";
+  }
+
+  identity ipip6 {
+    base ietf-if:interface-type;
+    description
+      "IPIP6 Tunnel.";
+  }
+
+  identity ipgre {
+    base ietf-if:interface-type;
+    description
+      "GRE over IP.";
+  }
+
+  typedef interface-ref {
+    type leafref {
+      path "/frr-interface:lib/frr-interface:interface/frr-interface:name";
+      require-instance false;
+    }
+    description
+      "Reference to an interface";
+  }
+
+  typedef if-flags-type {
+    type enumeration {
+      enum "up" {
+        value 1;
+        description
+          "Active and ready to transfer pkts.";
+      }
+      enum "broadcast" {
+        value 2;
+        description
+          "Broadcast enabled.";
+      }
+      enum "debug" {
+        value 3;
+        description
+          "Debug mode.";
+      }
+      enum "loopback" {
+        value 4;
+        description
+          "Loopback interface.";
+      }
+      enum "point-to-point" {
+        value 5;
+        description
+          "Point-to-Point link.";
+      }
+      enum "notrailers" {
+        value 6;
+        description
+          "BSD compatible.";
+      }
+      enum "running" {
+        value 7;
+        description
+          "Up and Running.";
+      }
+      enum "noarp" {
+        value 8;
+        description
+          "Can't perform ARP.";
+      }
+      enum "promisc" {
+        value 9;
+        description
+          "Promiscuous mode.";
+      }
+      enum "allmulti" {
+        value 10;
+        description
+          "Receive all multicast pkts.";
+      }
+      enum "simplex" {
+        value 11;
+        description
+          "Does not Rx or Tx the sametime.";
+      }
+      enum "link0" {
+        value 12;
+        description
+          "Link0";
+      }
+      enum "link1" {
+        value 13;
+        description
+          "Link1.";
+      }
+      enum "link2" {
+        value 14;
+        description
+          "Link2.";
```

Comment:
Really small nit, but `Link0` doesn't have a period and these two do, consistency would be good